packages/main/src/plugin/recommendations/recommendations-registry.ts

```diff
@@ -38,6 +38,12 @@ export class RecommendationsRegistry {
     private extensionsCatalog: ExtensionsCatalog,
   ) {}
 
+  isBannerRecommendationEnabled(): boolean {
+    return !this.configurationRegistry
+      .getConfiguration(RecommendationsSettings.SectionName)
+      .get<boolean>(RecommendationsSettings.IgnoreBannerRecommendations, false);
+  }
+
```

Comment:
issue: hello, on my end, I'm expecting that this method would be in charge of  checking both the specific flag and the global flag rather than having to do the check inside the getExtensionBanners method

Reply:
Updated